### ISSUE TITLE
strange bug, two reverse conclusion are shown out at the same run

### ISSUE THREAD

**Issue by dazauzai** [nf-core/sarek#1732]
### Description of the bug

Idk why, If I try to delete the two tools which report error, it will report for another problem
 -- Check script '/root/.nextflow/assets/nf-core/sarek/main.nf' at line: 342 or see '.nextflow.log' file for more details
I alter my samplesheet for many version. I have tried the sample sheet which only contain normal, tumour, and both. However I still face the same situation whatever I put in the sample sheet. I think I wont because the sample sheet.
[samplesheet.csv](https://github.com/user-attachments/files/17759884/samplesheet.csv)

### Command used and terminal output

```console
/mnt/c/Users/zbjia/Documents/GitHub/sarek# nextflow run nf-core/sarek    -profile docker    --input samplesheet.csv    --outdir /mnt/c/Users/zbjia/Documents/GitHub/sarek/work --wes --intervals /mnt/e/yiming/working/files/bed_file/V3_IDT_03861023_CML_Ph_v3_hg38.bed --tools ascat,cnvkit,controlfreec --fasta /mnt/e/yiming/working/fil
es/reference/GRCh38_masked_v2_decoy_gene.fasta --download_cache -r 3.4.4
 -- Check script '/root/.nextflow/assets/nf-core/sarek/main.nf' at line: 342 or see '.nextflow.log' file for more details
The sample-sheet only contains normal-samples, but the following tools, which were requested with "--tools", expect at least one tumor-sample : ascat, controlfreec
The sample-sheet only contains tumor-samples, but the following tools, which were requested by the option "tools", expect at least one normal-sample : ascat
```

**Comment by asp8200** [comment]
I think you may have got a case of [this](https://github.com/nf-core/sarek/issues/1629). I would ignore the error messages relating to the sample-sheet and try to look for another error message in `.nextflow.log`.

**Comment by FriederikeHanssen** [comment]
Which nextflow version are you using? If you are using 24.10 can you downgrade to the previous stable release?

**Comment by dazauzai** [comment]
> I think you may have got a case of [this](https://github.com/nf-core/sarek/issues/1629). I would ignore the error messages relating to the sample-sheet and try to look for another error message in `.nextflow.log`.

Yes, youre right. I guess the problem is because the problem Missing process or function Channel.empty([[]])
that is strange, I dont know this kind of code,  Channel.empty([[]]) why so many [][]

**Comment by dazauzai** [comment]
> Which nextflow version are you using? If you are using 24.10 can you downgrade to the previous stable release?

I tried. I have downgrade it to 24.09.1, But the bug still exsists.

**Comment by dazauzai** [comment]
> 您使用的是哪个版本的 nextflow？如果您使用的是 24.10，您可以降级到以前的稳定版本吗？



> Which nextflow version are you using? If you are using 24.10 can you downgrade to the previous stable release?
I tried to downgrade it to NXF_VER=24.04.4. now it successfully worked. Thanks very much for your help and your reply. Close this report if you want.

in conclusion, The bug which say that only contain tumour sample or normal sample are not the really problem. The problem is because Channel.empty([[]])
to fix this bug, just downgrade the nextflow to 24.04.4()
just like
 NXF_VER=24.04.4 nextflow run nf-core/sarek 
thanks again for your help, and hope it will be helpful for whom face the same problem with me.
